### PR TITLE
#1109: Improve Pit Coverage

### PR DESCRIPTION
**Pull request by goldbal330**
Signed-off-by: Alexander Goldberg <alexander.goldberg@bnymellon.com>

### CODE REVIEW

**Review by donraab — CHANGES_REQUESTED** [review]
Thanks for the contribution @goldbal330. A few changes requested.

File: eclipse-collections-code-generator/src/main/resources/test/utility/lazyPrimitiveIterateTest.stg [hunk 1]
```diff
@@ -40,6 +43,19 @@ public class Lazy<name>IterateTest
...
+    public void empty()
+    {
+        Assert.assertTrue(Lazy<name>Iterate.empty().isEmpty());
+    }
+
+    @Test
+    public void tap()
+    {
+        Lazy<name>Iterable <type>Iterable = Lazy<name>Iterate.tap(this.iterable, (<name>Procedure) each -> { });
+        Assert.assertEquals(this.iterable, <name>Lists.mutable.ofAll(<type>Iterable));
```
[the remark this answers is not in the document]
Comment (goldbal330) [reply eclipse-collections-code-generator/src/main/resources/test/utility/lazyPrimitiveIterateTest.stg:59]: Fixed!

File: eclipse-collections-code-generator/src/main/resources/test/utility/singletonPrimitiveIteratorTest.stg [hunk 2]
```diff
@@ -0,0 +1,61 @@
+import "copyright.stg"
```
[the remark this answers is not in the document]
Comment (goldbal330) [reply eclipse-collections-code-generator/src/main/resources/test/utility/singletonPrimitiveIteratorTest.stg]: Replaced.

File: unit-tests/src/test/java/org/eclipse/collections/impl/block/function/primitive/BooleanCaseFunctionTest.java [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2021 Goldman Sachs.
+ * Copyright (c) 2022 Goldman Sachs and others.
```
[the remark this answers is not in the document]
Comment (goldbal330) [reply unit-tests/src/test/java/org/eclipse/collections/impl/block/function/primitive/BooleanCaseFunctionTest.java:2]: Done!

File: unit-tests/src/test/java/org/eclipse/collections/impl/utility/primitive/LazyBooleanIterateTest.java [hunk 4]
```diff
@@ -30,4 +32,17 @@ public void collectIf()
...
+    public void empty()
+    {
+        Assert.assertTrue(LazyBooleanIterate.empty().isEmpty());
+    }
+
+    @Test
+    public void tap()
+    {
+        LazyBooleanIterable booleanIterable = LazyBooleanIterate.tap(this.iterable, (BooleanProcedure) each -> { });
+        Assert.assertEquals(this.iterable, BooleanLists.mutable.ofAll(booleanIterable));
```
[the remark this answers is not in the document]
Comment (goldbal330) [reply unit-tests/src/test/java/org/eclipse/collections/impl/utility/primitive/LazyBooleanIterateTest.java:48]: done!

**Review by donraab — APPROVED** [review]
LGTM. Thanks for the contribution @goldbal330 !

**Pull request merged**